source/agora/consensus/Validation.d

```diff
@@ -956,6 +957,53 @@ public bool isValidEnrollment (const ref Enrollment enrollment,
     return isInvalidEnrollmentReason(enrollment, block_height, findUTXO) is null;
 }
 
+/*******************************************************************************
+
+    Check the validity of a pre-image information
+
+    Pre-image infomation is considered valid if:
+        - The keys for enrollment in two pieces of pre-image information are
+            same
+        - Height for a pre-image is greater than height for a current image
+        - A current image is same as n times hashed value of a pre-image
+            (n = difference of two heights).
+
+    Params:
+        pre_image = The pre-image information
+        cur_image = The current image information
+
+    Returns:
+        `null` if the pre-image is valid, otherwise a string
+        explaining the reason it is invalid.
+
+*******************************************************************************/
+
+public string isInvalidPreimageReason (const ref PreimageInfo pre_image,
+    const ref PreimageInfo cur_image) nothrow @safe
+{
+    if (pre_image.enroll_key != cur_image.enroll_key)
+        return "The pre-image has a wrong enrollment key";
+
+    if (pre_image.height <= cur_image.height)
+        return "the pre-image has a invalid height";
+
+    Hash temp_hash = pre_image.hash;
+    for (uint i = 0; i < pre_image.height - cur_image.height; i++)
```

Comment:
What happens when a byzantine node sends a `PreimageInfo` with `pre_image.height` set to `ulong.max`?